### ISSUE TITLE
I would like to see a option to enable extra octave support in Noteblock API

### ISSUE THREAD

**Issue by chengts95** [Spliterash/MusicBox#10]
Since the extra octave can improve the nbs music a lot.

**Comment by Spliterash** [comment]
My plugin use noteblockapi for parse and play music, so i think if noteblock api can play extra octave, then my plugin can too

**Comment by chengts95** [comment]
> My plugin use noteblockapi for parse and play music, so i think if noteblock api can play extra octave, then my plugin can too

Yes but by default the the extra octave is disabled since it requires extra resource pack. So plugin needs to enable it. Please take a look at [this](https://github.com/SkytAsul/JukeBox/commit/b571a78d6eb5ca3ceb892667740c9224be27937a).

**Comment by joex92** [comment]
> Since the extra octave can improve the nbs music a lot.

actually that's a Minecraft limitation... not a plugin limitation... obviously you could add more octaves with resource packs, because you would be adding the octaves exaternally, but without adding something extra to both sides (client and server) you'll never get more than the octaves Minecraft already gives...

**Comment by chengts95** [comment]
yes, but there is an extra resource pack by OpenNBS and NoteBlock API itself has a selective resource pack for more octaves. That is why the  Noteblock API makes it an option. I have tested the extra octaves and it is awesome!  [this](https://github.com/SkytAsul/JukeBox/commit/b571a78d6eb5ca3ceb892667740c9224be27937a) mod is the only one to expose the API.

**Comment by joex92** [comment]
well first of all is not a mod, is a plugin... but I get it... I believe that mainly server owners prefer their servers to not make players download anything extra to be able to enter the server, otherwise just have a modded server instead of plugins... so I believe that's the reason, but it's true, it wouldn't hurt anyone to support the resource pack... and this could be the first one to do it, at least the first free plugin to do it...

**Comment by chengts95** [comment]
the server-side resource pack is a native MC functionality and many plugins are using it for custom items. And all I suggest is to have a better audio experience with these plugins, especially when the NoteBlock API already has this option.

**Comment by joex92** [comment]
well... the server-side thing is not just server-side... when the server sets a resource pack, it sends it to the client joining and, personally, sometimes that is annoying, specially if you are connecting from a low end PC, the processing of downloading and loading the RP takes too much time sometimes, and on some few cases the server can kick you for taking too much time at login... that's something server owners have to keep in mind... there are workarounds, like WorldResourcePack plugin, that lets you send the RP with a delay of *X* ticks...

Nevertheless, it would be good to add that functionality to the plugin, I haven't seen it in other free NBS plugin...

**Comment by Spliterash** [comment]
Resolved in PULL request #31
